pkg/run/install_dashboard.go

```diff
@@ -0,0 +1,293 @@
+package run
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta2"
+	wego "github.com/weaveworks/weave-gitops/api/v1alpha1"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	"github.com/weaveworks/weave-gitops/pkg/logger"
+	"github.com/weaveworks/weave-gitops/pkg/server"
+	"github.com/weaveworks/weave-gitops/pkg/utils"
+	"golang.org/x/crypto/bcrypt"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	helmRepositoryName      = "ww-gitops"
+	helmReleaseName         = "ww-gitops"
+	helmChartName           = "weave-gitops"
+	helmChartNamespacedName = wego.DefaultNamespace + "-ww-gitops"
+	podName                 = "ww-gitops-weave-gitops"
+	helmRepositoryUrl       = "https://helm.gitops.weave.works"
+)
+
+// InstallDashboard installs the GitOps Dashboard.
+func InstallDashboard(log logger.Logger, ctx context.Context, kubeClient *kube.KubeHTTP, kubeConfigArgs *genericclioptions.ConfigFlags) error {
+	log.Actionf("Installing the GitOps Dashboard ...")
+
+	password, err := utils.ReadPasswordFromStdin(log, "Please enter your password to generate the secret: ")
+	if err != nil {
+		log.Failuref("Could not read password")
+		return err
+	}
+
+	secret, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		log.Failuref("Error generating secret from password")
+		return err
+	}
+
+	log.Successf("Secret has been generated:")
+	fmt.Println(string(secret))
+
+	log.Actionf("Installing the GitOps Dashboard ...")
+
+	helmRepository := makeHelmRepository()
+	helmRelease, err := makeHelmRelease(log, string(secret))
+
+	if err != nil {
+		log.Failuref("Creating HelmRelease failed")
+		return err
+	}
+
+	manifests, err := generateManifests(log, string(secret), helmRepository, helmRelease)
+	if err != nil {
+		log.Failuref("Generating GitOps Dashboard manifests failed")
+		return err
+	}
+
+	log.Successf("Generated GitOps Dashboard manifests")
+
+	applyOutput, err := apply(log, ctx, kubeClient, kubeConfigArgs, manifests)
+	if err != nil {
+		log.Failuref("GitOps Dashboard install failed")
+		return err
+	}
+
+	log.Successf("GitOps Dashboard has been installed")
+
+	fmt.Println(applyOutput)
+
+	return nil
+}
+
+// IsDashboardInstalled checks if the GitOps Dashboard is installed.
+func IsDashboardInstalled(log logger.Logger, ctx context.Context, kubeClient *kube.KubeHTTP) bool {
```

Comment:
A small glitch here is that someone might install Weave GitOps with a different method.

But we can fix it later when needed.

Reply:
Would another method also require them to use the same HelmChart? Is it possible to install the dashboard without this HelmChart? Or will the chart be the same just its name will be different?

I mean if using a HelmChart is required for installing the dashboard can we just use something else (not the name) for detecting that this HelmChart is present reliably?